diff --git a/src/pipelines/canu/Execution.pm b/src/pipelines/canu/Execution.pm
index 4be46e73b..b0d6df076 100644
--- a/src/pipelines/canu/Execution.pm
+++ b/src/pipelines/canu/Execution.pm
@@ -1008,15 +1008,29 @@ sub buildGridJob ($$$$$$$$$) {
 
     open(F, "> $path/$script.jobSubmit-$idx.sh") or die;
     print F "#!/bin/sh\n";
+    print F "array=\"$arrayOpt\"\n";                             # insert 
+    print F "array=(\$array)\n";                                   #
+    print F "bgnLoop=\$(echo \"\${array[1]}\" | cut -d- -f1)\n"; # need to get the jobs submitted correctly
+    print F "endLoop=\$(echo \"\${array[1]}\" | cut -d- -f2)\n"; # use a loop instead of an array
     print F "\n";
+    print F "if [ -z \$bgnLoop ]\n";                             # insert
+    print F "then\n";                                            #
+    print F "bgnLoop=1\n";                                       # If only 1 job is going to run $arrayOpt
+    print F "endLoop=1\n";                                       # will be empty. Set loop [1..1]
+    print F "fi\n";                                              #
+    print F "for (( j=\$bgnLoop; j<=\$endLoop; j++ ))\n";
+    #print F "for j in {\$bgnLoop..\$endLoop}\n";                #insert, hopefully this will be 1..y
+    print F "do\n";                                              #insert
     print F "$submitCommand \\\n";
     print F "  $opts \\\n"  if (defined($opts));
     print F "  $nameOption \"$jobName\" \\\n";
-    print F "  $arrayOpt \\\n";
+    print F "  -v PBS_ARRAY_INDEX=\$j \\\n";             #insert
+    #print F "  $arrayOpt \\\n";                         #remove
     print F " -- " if (uc(getGlobal("gridEngine")) eq "PBSPRO");
     print F " -iscript_name:string=\"$script.sh\" -icanu_path:string=\"$path\" fetch_and_run \\\n" if (uc(getGlobal("gridEngine")) eq "DNANEXUS");
     print F "  ./$script.sh $arrayOff \\\n"                                                        if (uc(getGlobal("gridEngine")) ne "DNANEXUS");
     print F "> ./$script.jobSubmit-$idx.out 2>&1\n";
+    print F "done\n";                                  #insert
     close(F);
 
     makeExecutable("$path/$script.jobSubmit-$idx.sh");
